src/main/java/com/iota/iri/service/API.java

```diff
@@ -797,56 +807,40 @@ private AbstractResponse getBalancesStatement(final List<String> addrss, final L
         final List<Hash> addresses = addrss.stream().map(address -> (new Hash(address)))
                 .collect(Collectors.toCollection(LinkedList::new));
         final List<Hash> hashes;
+        final Map<Hash, Long> balances = new HashMap<>();
+        final int index = instance.milestone.latestSnapshot.index();
```

Comment:
accessed w/o holding an explicit lock.
while this may be ok (the lock is obtained inside Snapshot). this is hard to reason about.
better to just move the lock above it imo.